### PR TITLE
Do not use authority certificate for requests outside Che container

### PR DESCRIPTION
**Pull request by vitaliy-guliy**
Signed-off-by: admin <admin@admin.com>

### What does this PR do?
Disables using authority certificates when getting plugin list from custom plugin registry.

How to test:
- build and launch Che-Theia with changes from this PR
- open Plugins view, add registry (it could be https://raw.githubusercontent.com/vitaliy-guliy/che-theia-plugin-registry/master/plugins/plugins.json)

Screenshot from 2020-09-09 17-26-48

- after refreshing the view, scroll to bottom

Screenshot from 2020-09-09 17-32-20

Without this fix, new plugins will not appear in the list and Theia will trace the following
```
2020-07-30 13:01:29.028 root INFO Cannot access the registry Params: Error: unable to get local issuer certificate
    at TLSSocket.onConnectSecure (_tls_wrap.js:1501:34)
    at TLSSocket.emit (events.js:315:20)
    at TLSSocket._finishInit (_tls_wrap.js:936:8)
    at TLSWrap.ssl.onhandshakedone (_tls_wrap.js:710:12)
2020-07-30 13:01:29.148 root INFO Unable to read plugin registry Params: https://raw.githubusercontent.com/vitaliy-guliy/che-theia-plugin-registry/master/plugins/plugins.json
```

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/16473

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable

### CODE REVIEW

**Comment by ericwill** [comment]
> Signed-off-by: admin [admin@admin.com](mailto:admin@admin.com)

Looks like ECA check is failing because of this

**Review by azatsarynnyy — COMMENTED** [review]
I don't see how it should fix the mentioned issue https://github.com/eclipse/che/issues/16473 that's related to importing zipped project sources.
@vitaliy-guliy could you clarify it to me, please?

As I see, zip with project sources is imported by Workspace Plugin: https://github.com/eclipse/che-theia/blob/a321800c8359efc062b3820610087a7ac40cafbc/plugins/workspace-plugin/src/theia-commands.ts#L207

How it's related to the default plugin registry?

**Comment by vitaliy-guliy** [comment]
> I don't see how it should fix the mentioned issue [eclipse/che#16473](https://github.com/eclipse/che/issues/16473) that's related to importing zipped project sources.
> @vitaliy-guliy could you clarify it to me, please?
> 
> As I see, zip with project sources is imported by Workspace Plugin:
> 
> https://github.com/eclipse/che-theia/blob/a321800c8359efc062b3820610087a7ac40cafbc/plugins/workspace-plugin/src/theia-commands.ts#L207
> 
> How it's related to the default plugin registry?

This PR fixes a similar bug in Plugins service, which I mentioned in the issue https://github.com/eclipse/che/issues/16473#issuecomment-666424392

**Comment by vitaliy-guliy** [comment]
> > Signed-off-by: admin [admin@admin.com](mailto:admin@admin.com)
> 
> Looks like ECA check is failing because of this

Yes, indeed. It's because I made a commit from Che-Theia directly, and sometime I forget to setup the user email and name.
Thanks for noticing.

File: extensions/eclipse-che-theia-plugin-ext/src/node/che-plugin-service.ts [hunk 1]
```diff
@@ -107,12 +107,9 @@ export class ChePluginServiceImpl implements ChePluginService {
         }
     }
 
-    private getAxiosInstance(): AxiosInstance {
-        if (!this.isItNode()) {
-            return axios;
-        }
-
+    private getAxiosInstance(uri: string): AxiosInstance {
```
Comment (azatsarynnyy) [inline extensions/eclipse-che-theia-plugin-ext/src/node/che-plugin-service.ts:110]: It's hard to understand what `uri` parameter means here, without examining the method internals.
Please, document it or consider replacing it with a flag, e.g. `attachCerts: boolean`.

**Comment by azatsarynnyy** [comment]
> This PR fixes a similar bug in Plugins service, which I mentioned in the issue [eclipse/che#16473 (comment)](https://github.com/eclipse/che/issues/16473#issuecomment-666424392)

I see. Thanks!

**Comment by dmytro-ndp** [comment]
[ci-build]

**Comment by dmytro-ndp** [comment]
[crw-ci-test]

**Comment by dmytro-ndp** [comment]
crw-ci-test

File: extensions/eclipse-che-theia-plugin-ext/src/node/che-plugin-service.ts (hunk 1 above)
Comment (vitaliy-guliy) [reply extensions/eclipse-che-theia-plugin-ext/src/node/che-plugin-service.ts:114]: Done

File: extensions/eclipse-che-theia-plugin-ext/src/node/che-plugin-service.ts [hunk 2]
```diff
@@ -107,12 +107,13 @@ export class ChePluginServiceImpl implements ChePluginService {
         }
     }
 
-    private getAxiosInstance(): AxiosInstance {
-        if (!this.isItNode()) {
-            return axios;
-        }
-
+    /**
+     * URI for which an axios instance will be created.
```
Comment (azatsarynnyy) [inline extensions/eclipse-che-theia-plugin-ext/src/node/che-plugin-service.ts:111]: The returned axios instance isn't tied to a specific URI, so I would remove this sentence.

**Review by azatsarynnyy — APPROVED** [review]

File: extensions/eclipse-che-theia-plugin-ext/src/node/che-plugin-service.ts (hunk 2 above)
Comment (vitaliy-guliy) [reply extensions/eclipse-che-theia-plugin-ext/src/node/che-plugin-service.ts]: Done

**Pull request merged**